### ISSUE TITLE
Suggestion / Potential Enhancement -  Blockfrost Pools Load address's once For Assets & Dexs

### ISSUE THREAD

**Issue by dpbeaumont** [IndigoProtocol/dexter#25]
Suggestion - Is it possible that we could potentially add an option to get all the pools addresses for say 5 assets (MIN,AGIX,INDY etc for DEX's Minswap, Sundaeswap etc first, so our consuming application can re-use this data going forwards with potentially a get poolById going forwards for price querying per asset. This could help reduce requests to blockfrost if this is achievable and could potentially reduce the likelihood for autobans/rate limiting when calling out to get all the pools and filtering again. 

Would be good to hear thoughts on this! If it is achievable or not :)

**Comment by Sluder** [comment]
That's a good idea! Are you thinking maybe adding a function in the fetch request like `getPoolByAddress(poolAddress: string)`? Or maybe a function to get the latest pool state - `getLatestPoolState(pool: LIquidityPool)`

**Comment by dpbeaumont** [comment]
> That's a good idea! Are you thinking maybe adding a function in the fetch request like `getPoolByAddress(poolAddress: string)`? Or maybe a function to get the latest pool state - `getLatestPoolState(pool: LIquidityPool)`

Yeah was thinking exactly that. Something that can get the pool details on demand if required but only for the pool required. :)

**Comment by dpbeaumont** [comment]
Wanted to share some testing I did on this, with loading the pools as a one off application load and incorporating  a custom Blockfrost SDK data provider for the data access to enforce default Blockfrost rate limiting I could see the pools returning with an average of 2-4 minutes whilst rate limited.

I then attempted querying them going forwards on demand when required with the ones I stored by the address, with a refresh pool state method mechanism to get the latest prices, although the price information was not accurate. This did bring down the query time from 2-4 minutes on average, to results every 8 seconds.
